### PR TITLE
Updates and Reformating

### PR DESCRIPTION
**Pull request by KaNaDaAT**
- Arrays now have a keyword `:len` for length
- Arrays now can be used with variables as indexer
- _true_ and _false_ now work properly
- _funqtions_ can now return in the middle of them self and not jsut at the end
- Now supports `~?` as alias for _if_
- No longer has issues with variables from "higer" contexts
- _funqtions_ without parameters possible now
- _string_ can also be decleared empty now
- **Lib:** Better Console
- **Lib:** Random Number
- **Lib:** Parse String to int/decimal/double/bool
- Some Bugs Fixed

### CODE REVIEW

File: sqripts/loops.sq [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ﻿*~ n <~ 0;
 while (n < 10) {
 	n <~ n + 1;
-	*~ t <~ "we're at step " + n + "
-	";
+	*~ t <~ "we're at step " + n + "\n";
```
Comment (qrakhen) [inline sqripts/loops.sq]: didn't know that worked yet, lol

Comment (KaNaDaAT) [reply sqripts/loops.sq]: Cause I added it :P haha

File: Core/Digesters/Funqtionizer.cs [hunk 2]
```diff
@@ -2,62 +2,84 @@
...
-        public static Value[] parseParameters(Qontext context, Token[] stack) {
-            return new Funqtionizer(stack).parseParameters(context);
-        }
-    }
-
-    internal class FunqtionizerException : Exception
-    {
-        public FunqtionizerException(string message, Token cause = null) : base(message, cause) { }
-    }
+
```
Comment (qrakhen) [inline Core/Digesters/Funqtionizer.cs:5]:
```suggestion
```

File: sqripts/Testing.sq [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	console:write("Congratulations:", "DarkBlue");
+	console:writeln(currentPlayerName, "Blue");
+	console:writeln("----------------------\n\n", "DarkBlue");
+	
+	console:write('Again? [y/n]: ');
+	*~ input <~ console:read();
+	~? ((input != 'y') && (input != 'Y')) {
+		run <~ 0;
+	}
+}
```
Comment (qrakhen) [inline sqripts/Testing.sq:225]: i don't think we need the testing sqript here, but i still appreciate the contribution :D

File: sqripts/backup.sq [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	*~ c <~ 0;
+	loop (index < listParam:len) {
+		loop (index < listParam:len) {
+			console:write(listParam:index:0 + " ");
+			index <~ index + 1;
+		}
+	}
+});
+
+printList(emptyboard);
```
Comment (qrakhen) [inline sqripts/backup.sq]: what?! [[1],[2]] works? holy shit!

File: sqripts/Testing.sq (hunk 3 above)
Comment (KaNaDaAT) [reply sqripts/Testing.sq:225]: Ah yeah the Testing.sq was the one executing automatically in VS as args param. You can remove that.

**Pull request merged**